keras/layers/containers.py

```diff
@@ -75,6 +75,19 @@ def updates(self):
         return updates
 
     @property
+    def state_updates(self):
+        state_updates = []
+        for l in self.layers:
+            if l.trainable and getattr(l, 'stateful', False):
```

Comment:
Unclear why it should be restricted to trainable layers. Having this behavior in any stateful layer is fine imo.